smart-py/marketplace_tests.py

```diff
@@ -0,0 +1,870 @@
+    manager = sp.test_account("manager")
+    objkt = sp.test_account("objkt123")
+
+    creator = sp.test_account("creator")
+    # TODO how to turn into the binary? or is the binary a pointer?
+    metadata = sp.record(
+        name = "test",
+        description = "test",
+        tags = [
+            'test'
+        ],
+        symbol = 'OBJKT',
+        artifactUri = "ipfs://test",
+        displayUri = "ipfs://test",
+        thumbnailUri = "ipfs://test",
+        creators = [
+            creator.address
+        ],
+        formats = [
+            {
+                "uri":"ipfs://test",
+                "mimeType":"image/png"
+            }
+        ],
+        decimals = 0,
+        isBooleanAmount = False,
+        shouldPreferSymbol = False
+    )
+
+    fee = 25
+
+    swap = Marketplace(
+        objkt.address,
+        metadata,
+        manager.address,
+        fee
+    )
+
+    scenario += swap
+
+    scenario.verify(swap.data.fee == 25)
+
+    # illegal attempts (lateral)
+    scenario += swap.update_fee(3).run(sender = objkt.address, valid = False)
+    scenario.verify(swap.data.fee == 25)
+
+    scenario += swap.update_fee(3).run(sender = seller.address, valid = False)
+
+    scenario.verify(swap.data.fee == 25)
+
+    # valid for manager to change the fee
+    scenario += swap.update_fee(3).run(sender = manager.address, valid = True)
+
+    scenario.verify(swap.data.fee == 3)
+
+@sp.add_test("Test swap")
+def test():
+    # init test and create html output
+    scenario = sp.test_scenario()
+    scenario.h1("Swap Test")
+
+    # init test values
+    seller = sp.test_account("seller")
+    manager = sp.test_account("manager")
+    objkt = sp.test_account("objkt123")
+
+    creator = sp.test_account("creator")
+    metadata = sp.record(
+        name = "test",
+        description = "test",
+        tags = [
+            'test'
+        ],
+        symbol = 'OBJKT',
+        artifactUri = "ipfs://test",
+        displayUri = "ipfs://test",
+        thumbnailUri = "ipfs://test",
+        creators = [
+            creator.address
+        ],
+        formats = [
+            {
+                "uri":"ipfs://test",
+                "mimeType":"image/png"
+            }
+        ],
+        decimals = 0,
+        isBooleanAmount = False,
+        shouldPreferSymbol = False
+    )
+
+    fee = 25
+
+    swap = Marketplace(
+        objkt.address,
+        metadata,
+        manager.address,
+        fee
+    )
+
+    scenario += swap
+
+    # no swaps yet
+    scenario.verify(swap.data.counter == 500000)
+    scenario.verify(swap.data.swaps.contains(500000) == False)
+
+    # swap with objkt id above max must fail
+    # there are no objkts at this stage so should fail
+    #
+    # TODO
+    #
+    # This should fail but it passes, however i don't know if i'm
+    # making the objkt in the same way i think i'm making the objkt
+    # scenario += swap.swap(
+    #     creator=creator.address,
```

Comment:
Nit. Inconsistent spacing: creator = creator.address,
Also, royalties next line and other tests.

Reply:
👍